addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -72,7 +73,10 @@ options.registry.TableOfContent = options.Class.extend({
         // We dispose the scrollSpy because the navbar will be updated.
         this._disposeScrollSpy();
         const $nav = this.$target.find('.s_table_of_content_navbar');
-        const $headings = this.$target.find(this.targetedElements);
+        // As the navbar is hidden in mobile view, we only consider the headings
+        // visible on desktop.
+        const $headings = this.$target.find(this.targetedElements).toArray()
+            .filter(el => !el.closest('.o_snippet_desktop_invisible'));
```

Comment:
If I am not mistaken, after this, `$headings` is not a jQuery object anymore and should thus not be named that way anymore.